phi/tools/giphy.py

```diff
@@ -0,0 +1,70 @@
+import uuid
+from typing import Optional
+import httpx
+import os
+
+from phi.agent import Agent
+from phi.model.content import Image
+from phi.tools import Toolkit
+from phi.utils.log import logger
+
+
+class GiphyTools(Toolkit):
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        limit: int = 1,
+    ):
+        super().__init__(name="giphy_tools")
+
+        self.api_key = api_key or os.getenv("GIPHY_API_KEY")
+        if not self.api_key:
+            logger.error("No Giphy API key provided")
+
+        self.limit: int = limit
+
+        self.register(self.search_gifs)
+
+    def search_gifs(self, agent: Agent, query: str) -> str:
+        """Find a GIPHY gif
+
+        Args:
+            query (str): A text description of the required gif.
+
+        Returns:
+            The resulting gif.
```

Comment:
result (str): A string containing urls of GIFs found

Reply:
This is just for added context for the model, so it can format the URLs in a organized way